src/gui/dialogues/newpilotdialog.cpp

```diff
@@ -149,7 +149,7 @@ void NewPilotDialog::submitForm()
     DEB("Creating Database Object...");
     QMap<QString, QString> newData;
 
-    auto line_edits = parent()->findChildren<QLineEdit *>();
+    auto line_edits = this->findChildren<QLineEdit *>();
 
     for (const auto &le : line_edits) {
```

Comment:
`this->findChildren` Ensure that we have the correct line edits. The dialog *only* has the necessary information. Therefore the for loop can be simplyfied into what i had in the previous pull request #13.
```c++
for(auto& le : line_edits) {
    auto key = le->objectName().remove("LineEdit") // revove substring "LineEdit" and return new QString
    auto key = le->text();
    newData.insert(key, value);  // no need to check the key because it *will* exist and be valid
}
```

Reply:
thanks, I missed that. Included in new commit.